tests/test-st2-invalid-auth.js

```diff
@@ -0,0 +1,168 @@
+    // https://medium.com/@gattermeier/invalidate-node-js-require-cache-c2989af8f8b0
+    delete require.cache[require.resolve("../scripts/stackstorm.js")];
+  });
+
+  it("should raise exception with null auth URL", function(done) {
+    // Mock process.env for all modules
+    // https://glebbahmutov.com/blog/mocking-process-env/
+    restore_env = mockedEnv({
+      ST2_AUTH_USERNAME: 'nonexistent-st2-auth-username',
+      ST2_AUTH_PASSWORD: 'nonexistent-st2-auth-password'
+    });
+
+    // Load script under test
+    var stackstorm = require("../scripts/stackstorm.js");
+    stackstorm(robot).then(function (stop) {
+      expect(error_spy).to.have.been.calledOnce;
+      expect(error_spy.firstCall.args[0]).include('Environment variables ST2_AUTH_USERNAME, ST2_AUTH_PASSWORD and ST2_AUTH_URL should only be used together.');
+      stop({shutdown: true});
+
+      done();
+    }).catch(function (err) {
+      console.log(err);
+      done(err);
+    });
+  });
+
+  it("should raise exception with bad auth URL", function(done) {
+    restore_env = mockedEnv({
+      ST2_AUTH_URL: 'https://nonexistent-st2-auth-url:9101',
+      ST2_AUTH_USERNAME: 'nonexistent-st2-auth-username',
+      ST2_AUTH_PASSWORD: 'nonexistent-st2-auth-password'
+    });
+
+    // Load script under test
+    var i, stackstorm = require("../scripts/stackstorm.js");
+    stackstorm(robot).then(function (stop) {
+      expect(error_spy.args).to.have.lengthOf.above(2);
+
+      // Check that it was called at some point with 'Failed to retrieve commands from'
+      for (i = 0; i < error_spy.args.length; i++) {
+        try {
+          expect(error_spy.args[i][0]).to.be.a('string');
+          expect(error_spy.args[i][0]).to.startWith('Failed to retrieve commands from');
+          break;
+        } catch (err) {
+          // If we have reached the last call and we still haven't found it
+          if (i >= error_spy.args.length-1) {
+            // Re-throw the assert exception
+            throw(err);
+          }
+          // Implicit continue
+        }
+      }
+
+      // Check that it was called at some point with 'Failed to retrieve commands from'
+      for (i = 0; i < error_spy.args.length; i++) {
+        try {
+          expect(error_spy.args[i][0]).to.be.a('string');
+          expect(error_spy.args[i][0]).to.include('getaddrinfo');
+          expect(error_spy.args[i][0]).to.include('nonexistent-st2-auth-url nonexistent-st2-auth-url:9101');
+          break;
+        } catch (err) {
+          // If we have reached the last call and we still haven't found it
+          if (i >= error_spy.args.length-1) {
+            // Re-throw the assert exception
+            throw(err);
+          }
+          // Implicit continue
+        }
+      }
+
+      stop({shutdown: true});
+
+      done();
+    }).catch(function (err) {
+      console.log(err);
+      done(err);
+    });
+  });
+
+  it("should raise exception with bad API key", function(done) {
```

Comment:
Is this test description still actual?

As @jinpingh found, bad API key doesn't lead to uncaught exception and end of bot's life.